### PR TITLE
Feature/android media session

### PR DESCRIPTION
**Pull request by matejpolak**
#### Describe the changes
Added mediaSession for Android ExoPlayer to let device know about currently used player. With this in place you can use some native features like pause/play using HW buttons on any compatible headset connected to the device, displaying playback buttons in PIP window (previous, play/pause, next), etc.. 

Currently using the simplest solution with usage of external ExoPlayer mediaSession extension library, which takes care of all needed mediaSession callback, and the session it self. To support more advanced features, full implementation may be needed. 

Since I'm not an Android expert, it's possible that some things, or additional metadata are missing. Please consider this as MVP which is currently implemented mainly to support PIP player controls. Feel free to suggest any change or pick deeper implementation if needed.

Also added `onExternalPauseToggled ` event prop, which notifies client about new play/pause state, so JS can perform needed actions based on this information. For example change UI state.

### CODE REVIEW

File: android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java [hunk 1]
```diff
@@ -62,6 +63,8 @@
 import com.google.android.exoplayer2.upstream.DefaultAllocator;
 import com.google.android.exoplayer2.upstream.DefaultBandwidthMeter;
 import com.google.android.exoplayer2.util.Util;
+import com.google.android.exoplayer2.ui.PlayerControlView;
```
Comment (CHaNGeTe) [inline android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java]: this import is duplicated

**Review by CHaNGeTe — CHANGES_REQUESTED** [review]

File: android-exoplayer/src/main/java/com/brentvatne/exoplayer/VideoEventEmitter.java [hunk 2]
```diff
@@ -193,6 +197,12 @@ void ready() {
         receiveEvent(EVENT_READY, null);
     }
 
+    void externalPauseTriggered(boolean isPlaying) {
```
Comment (CHaNGeTe) [inline android-exoplayer/src/main/java/com/brentvatne/exoplayer/VideoEventEmitter.java]: the name of the func should'nt be toggled as well?

**Review by CHaNGeTe — APPROVED** [review]

**Review by AurangzaibRamzan — APPROVED** [review]

**Comment by rogerkerse** [comment]
Hi, what is the state with this pull request? 😊 Would be very grateful if it got merged. We are using picture in picture on Android through another library, but since current version of react-native-video doesn't have media session, there are no playback buttons :(

**Comment by EyMaddis** [comment]
Just a heads up if this ever gets merged:
`onExternalPauseToggled` is not necessary, there is already the prop `onPlaybackRateChange` that should be able to do the same. I just stumbled onto this PR, I did not test it.

**Pull request closed without merging**